Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2006-01-13
Initial Package Version: 2.6.15
Origin: Jim Gifford
Upstream Status: Unknown
Description: Fixes Build issue on Sparc Pure 64
 
diff -Naur linux-2.6.15.orig/arch/sparc64/kernel/process.c linux-2.6.15/arch/sparc64/kernel/process.c
--- linux-2.6.15.orig/arch/sparc64/kernel/process.c	2006-01-03 03:21:10.000000000 +0000
+++ linux-2.6.15/arch/sparc64/kernel/process.c	2006-01-13 19:27:51.000000000 +0000
@@ -164,6 +164,7 @@
 	panic("Reboot failed!");
 }
 
+#ifdef CONFIG_COMPAT
 static void show_regwindow32(struct pt_regs *regs)
 {
 	struct reg_window32 __user *rw;
@@ -189,6 +190,7 @@
 	       r_w.ins[0], r_w.ins[1], r_w.ins[2], r_w.ins[3],
 	       r_w.ins[4], r_w.ins[5], r_w.ins[6], r_w.ins[7]);
 }
+#endif
 
 static void show_regwindow(struct pt_regs *regs)
 {
@@ -214,7 +216,11 @@
 			set_fs (old_fs);			
 		}
 	} else {
-		show_regwindow32(regs);
+		#ifdef CONFIG_COMPAT
+			show_regwindow32(regs);
+		#else
+			show_regwindow(regs);
+		#endif
 		return;
 	}
 	printk("l0: %016lx l1: %016lx l2: %016lx l3: %016lx\n",
diff -Naur linux-2.6.15.orig/arch/sparc64/kernel/sparc64_ksyms.c linux-2.6.15/arch/sparc64/kernel/sparc64_ksyms.c
--- linux-2.6.15.orig/arch/sparc64/kernel/sparc64_ksyms.c	2006-01-03 03:21:10.000000000 +0000
+++ linux-2.6.15/arch/sparc64/kernel/sparc64_ksyms.c	2006-01-13 19:28:44.000000000 +0000
@@ -239,7 +239,9 @@
 
 /* Solaris/SunOS binary compatibility */
 EXPORT_SYMBOL(_sigpause_common);
+#ifdef CONFIG_COMPAT
 EXPORT_SYMBOL(verify_compat_iovec);
+#endif
 
 EXPORT_SYMBOL(dump_thread);
 EXPORT_SYMBOL(dump_fpu);
